### PR TITLE
Store seen cards' links in localstorage

### PR DESCRIPTION
**Pull request by jasonklein**
This PR stores the seen content items (cards) in the browser's localstorage.

This will eventually be changed to instead or additionally store which content items' games/quizzes have been passed.

### CODE REVIEW

File: src/components/Card/CardList.test.tsx [hunk 1]
```diff
@@ -48,6 +48,10 @@ describe("CardList component", () => {
     );
   });
 
+  it("adds a card link to localstorage when card item is swiped", () => {
```
Comment (edmondtam1) [inline src/components/Card/CardList.test.tsx:51]: ChatGPT will help write this for you 😅

File: src/components/Card/CardList.tsx [hunk 2]
```diff
@@ -55,7 +68,7 @@ export function CardList({ cardItemData }: CardListProps) {
         <CardItem
           key={card.link}
           data={card}
-          handleSwipe={(dir) => swiped(dir, card.title)}
+          handleSwipe={(dir) => swiped(dir, card.title, card.link)}
```
Comment (edmondtam1) [inline src/components/Card/CardList.tsx:71]: Really sorry I have adjusted this to be `const swiped = (direction: string, card: CardItemData) => {}` now, if you look at the other PR

**Review by edmondtam1 — CHANGES_REQUESTED** [review]

File: src/components/Card/CardList.test.tsx (hunk 1 above)
Comment (jasonklein) [reply src/components/Card/CardList.test.tsx:69]: The tests ChatGPT offered were not quite right. I think I need to sit and properly go through a React testing tutorial. I will try to make time for that next week.

Comment (edmondtam1) [reply src/components/Card/CardList.test.tsx:69]: ! I'm surprised. It gave me helpful structures and methods that I wouldn't have known about, and it was just a matter of integrating it with what I kinda knew already. I haven't been writing many React tests though so would love to see how you approach this.

**Comment by edmondtam1** [comment]
Very interesting! I'll have to look closely at the code, but I had just one thought from trying out the deploy preview - can we store everything that they interact with? Maybe these "actions" to start:
* Completed (quiz is done correctly)
* Attempted (quiz is done incorrectly)
* Selected (swiped right and quiz is done)
* Passed (swiped left)

Or whatever you feel works. Tracking everything seems sensible to me, but will leave it to you.

Also I noticed that the array starts with `[""]` and then it appends the successfully completed ones `["", "kraken.link"]`, but I'll look closer at the code if needed.

**Comment by jasonklein** [comment]
We can and presumably will store some of that, but those all seem like additional features on top of the basic one that we should finish implementing first, so I don't think they belong in this PR.

In terms of the empty strings in the array, that's because when `useEffect` runs initially in production, `lastCompletedCardLink` is an empty string. Also, when it runs in development, it does a test run (so you will see two empty strings). But I've stopped being lazy and added a clean-up function that takes care of that.

If this looks good to you (even though tests are still missing), LMK and I will merge it in.
